xonsh/base_shell.py

```diff
@@ -357,7 +357,10 @@ def singleline(self, **kwargs):
 
     def precmd(self, line):
         """Called just before execution of line."""
-        self.precwd = os.getcwd()
+        try:
+            self.precwd = os.getcwd()
+        except FileNotFoundError as f:
+            self.precwd = "/"
```

Comment:
Instead of pointing to root path, we can use user's home directory.  Also it needs to work on windows as well. 

```
from os.path import expanduser
home = expanduser("~")
```

Reply:
sure. Thank You. Implementing.